driver/interpolate_driver.hpp

```diff
@@ -0,0 +1,554 @@
+#include "timer.hpp"
+#include "util_driver.hpp"
+
+#include <../test/tensor_holder.hpp>
+#include <../test/verify.hpp>
+
+#include <miopen/env.hpp>
+#include <miopen/handle.hpp>
+#include <miopen/miopen.h>
+#include <miopen/tensor.hpp>
+#include <vector>
+
+inline std::vector<int> GetStrides(std::vector<int> lengths, int contiguous)
+{
+    if(contiguous != 0 && contiguous != 1)
+        std::cerr << "Error Tensor Contiguous should be 0 or 1" << std::endl;
+    if(contiguous == 0)
+        std::swap(lengths.front(), lengths.back());
+    std::vector<int> strides(lengths.size());
+    strides.back() = 1;
+    for(int i = lengths.size() - 2; i >= 0; --i)
+        strides[i] = strides[i + 1] * lengths[i + 1];
+    if(contiguous == 0)
+        std::swap(strides.front(), strides.back());
+    return strides;
+}
+
+template <typename Tgpu, typename Tref>
+class InterpolateDriver : public Driver
+{
+public:
+    InterpolateDriver() : Driver()
+    {
+        miopenCreateTensorDescriptor(&inputDesc);
+        miopenCreateTensorDescriptor(&outputDesc);
+        miopenCreateTensorDescriptor(&outputGradDesc);
+        miopenCreateTensorDescriptor(&inputGradDesc);
+        miopenCreateTensorDescriptor(&scaleFactorsDesc);
+
+        data_type = miopen_type<Tgpu>{};
+    }
+
+    int AddCmdLineArgs() override;
+    int ParseCmdLineArgs(int argc, char* argv[]) override;
+    InputFlags& GetInputFlags() override { return inflags; }
+
+    template <typename T>
+    std::vector<T> GetTensorFromCmd(const char* param);
+    int GetandSetData() override;
+
+    int AllocateBuffersAndCopy() override;
+
+    int RunForwardGPU() override;
+    int RunForwardCPU();
+
+    int RunBackwardGPU() override;
+    int RunBackwardCPU();
+
+    int VerifyBackward() override;
+    int VerifyForward() override;
+    ~InterpolateDriver() override
+    {
+        miopenDestroyTensorDescriptor(inputDesc);
+        miopenDestroyTensorDescriptor(outputDesc);
+        miopenDestroyTensorDescriptor(outputGradDesc);
+        miopenDestroyTensorDescriptor(inputGradDesc);
+        miopenDestroyTensorDescriptor(scaleFactorsDesc);
+    }
+
+private:
+    InputFlags inflags;
+
+    int forw;
+
+    miopenTensorDescriptor_t inputDesc;
+    miopenTensorDescriptor_t outputDesc;
+    miopenTensorDescriptor_t outputGradDesc;
+    miopenTensorDescriptor_t inputGradDesc;
+    miopenTensorDescriptor_t scaleFactorsDesc;
+
+    std::unique_ptr<GPUMem> in_dev;
+    std::unique_ptr<GPUMem> out_dev;
+    std::unique_ptr<GPUMem> out_grad_dev;
+    std::unique_ptr<GPUMem> in_grad_dev;
+    std::unique_ptr<GPUMem> scale_factors_dev;
+    std::unique_ptr<GPUMem> workspace_dev;
+
+    std::vector<Tgpu> in;
+    std::vector<Tgpu> out;
+    std::vector<Tref> out_host;
+
+    std::vector<float> scale_factors;
+
+    std::vector<Tgpu> out_grad;
+    std::vector<Tgpu> in_grad;
+    std::vector<Tref> in_grad_host;
+    std::vector<float> workspace;
+
+    std::vector<int> in_len;
+    std::vector<int> size;
```

Comment:
uint64_t. and you should also fix related things below

Reply:
why should I use uint64_t, not int?

Reply:
Because int is 32bit or 64bit depend on architecture. In this context in_len elements should always be 64-bit

Reply:
<img width="550" alt="image" src="https://github.com/user-attachments/assets/63314bab-1f9e-4a4d-88cb-c711bb8fc88c">
Not sure about that, 32bit or 64bit is all enough range for in_len

Reply:
ok